src/state/farms/index.ts

```diff
@@ -27,6 +29,22 @@ const noAccountFarmConfig = farmsConfig.map((farm) => ({
 
 const initialState: FarmsState = { data: noAccountFarmConfig, loadArchivedFarmsData: false, userDataLoaded: false }
 
+// Async thunks
+
+export const fetchFarmsData = createAsyncThunk('farms/fetchFarmsData', async () => {
+  const farmsData = await fetchFarms(nonArchivedFarms)
+  return farmsData
+})
+
+export const fetchFarmData = createAsyncThunk<{ pid: number; data: Farm }, number>(
+  'farms/fetchFarmData',
+  async (pid) => {
+    const farm = farmsConfig.find((farmConfig) => farmConfig.pid === pid)
```

Comment:
I chose to get the farm data from the config instead of passing it in because we wouldn't be able to use it in any hooks as the object is mutating all the time causing re-render.